Cilium segfaults when L7 visibility is enabled in conjuction with querying Hubble server using tcp-flag filter

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

1. Enable L7 visibility for a pod (add `io.cilium.proxy-visibility` annotation)
2. Enable Hubble and run hubble-observe with `tcp-flags` filter (`hubble obseve -f --tcp-flags syn`)
3. Send some HTTP traffic
4. `cilium-agent` which is responsible for providing the L7 visibility will crash due to segfault.

### Cilium Version

v1.11.1

### Kernel Version

N/A

### Kubernetes Version

N/A

### Sysdump

N/A

### Relevant log output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x29 pc=0x1d145b5]

goroutine 21447 [running]:
github.com/cilium/cilium/pkg/hubble/filters.filterByTCPFlags.func1(0x0)
        /go/src/github.com/cilium/cilium/pkg/hubble/filters/tcp.go:27 +0x95
github.com/cilium/cilium/pkg/hubble/filters.FilterFuncs.MatchAll({0xc00221e510, 0x1, 0xc002d12702}, 0xc000c802a0)
        /go/src/github.com/cilium/cilium/pkg/hubble/filters/filters.go:29 +0x56
github.com/cilium/cilium/pkg/hubble/filters.BuildFilterList.func1(0xc001760618)
        /go/src/github.com/cilium/cilium/pkg/hubble/filters/filters.go:116 +0x2b
github.com/cilium/cilium/pkg/hubble/filters.FilterFuncs.MatchOne({0xc002c55b80, 0x2, 0xc002d127c0}, 0xc0028391d0)
        /go/src/github.com/cilium/cilium/pkg/hubble/filters/filters.go:44 +0x76
github.com/cilium/cilium/pkg/hubble/filters.Apply({0xc002c55b80, 0x2c52a70, 0xc002d127c0}, {0x4872150, 0x0, 0x0}, 0xc003255978)
        /go/src/github.com/cilium/cilium/pkg/hubble/filters/filters.go:23 +0x35
github.com/cilium/cilium/pkg/hubble/observer.(*eventsReader).Next(0xc000c80240, {0x2c52a70, 0xc002d127c0})
        /go/src/github.com/cilium/cilium/pkg/hubble/observer/local_observer.go:567 +0x19d
github.com/cilium/cilium/pkg/hubble/observer.(*LocalObserverServer).GetFlows(0xc000434280, 0xc00244c580, {0x2c85710, 0xc002c55b60})
        /go/src/github.com/cilium/cilium/pkg/hubble/observer/local_observer.go:273 +0x5f5
github.com/cilium/cilium/api/v1/observer._Observer_GetFlows_Handler({0x2671860, 0xc000434280}, {0x2c77810, 0xc0012b8e40})
        /go/src/github.com/cilium/cilium/api/v1/observer/observer_grpc.pb.go:207 +0xd0
google.golang.org/grpc.(*Server).processStreamingRPC(0xc0017e56c0, {0x2c96a90, 0xc00135d680}, 0xc00289e120, 0xc000cbfce0, 0x42e6ae0, 0x0)
        /go/src/github.com/cilium/cilium/vendor/google.golang.org/grpc/server.go:1540 +0xeba
google.golang.org/grpc.(*Server).handleStream(0xc0017e56c0, {0x2c96a90, 0xc00135d680}, 0xc00289e120, 0x0)
        /go/src/github.com/cilium/cilium/vendor/google.golang.org/grpc/server.go:1620 +0x9e5
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /go/src/github.com/cilium/cilium/vendor/google.golang.org/grpc/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /go/src/github.com/cilium/cilium/vendor/google.golang.org/grpc/server.go:919 +0x294
```


### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/pkg/hubble/filters/tcp.go
+++ b/pkg/hubble/filters/tcp.go
@@ -13,11 +13,10 @@ import (
 
 func filterByTCPFlags(flags []*flowpb.TCPFlags) (FilterFunc, error) {
 	return func(ev *v1.Event) bool {
-		l4tcp := ev.GetFlow().GetL4().GetTCP()
-		if l4tcp == nil {
+		flowFlags := ev.GetFlow().GetL4().GetTCP().GetFlags()
+		if flowFlags == nil {
 			return false
 		}
-		flowFlags := l4tcp.GetFlags()
 		// check if the TCP event has any of the flags mentioned in flowfilter
 		// example: if TCP event has flags SYN and ACK set and if the flowfilter
 		// only has SYN, then this event should be accepted by the filter.

